helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -220,17 +221,24 @@ public GenericHelixController(String clusterName, Set<Pipeline.Type> enabledPipe
   class RebalanceTask extends TimerTask {
     final HelixManager _manager;
     final ClusterEventType _clusterEventType;
+    private final boolean _shouldRefreshCache;
     private long _nextRebalanceTime;
 
     public RebalanceTask(HelixManager manager, ClusterEventType clusterEventType) {
       this(manager, clusterEventType, -1);
+    }
 
+    public RebalanceTask(HelixManager manager, ClusterEventType clusterEventType,
```

Comment:
can we use a simple Builder pattern; It's package-private class so I assume the constructor will be called in very limited places.

Reply:
A Builder pattern would be overdesign here since this class is private.